rb/lib/selenium/webdriver/remote/w3c/bridge.rb

```diff
@@ -116,6 +116,20 @@ def page_source
                               'return source;')
           end
 
+          #
+          # Create a new top-level browsing context
+          # https://w3c.github.io/webdriver/#new-window
+          # @param type [String] Supports two values: 'tab' and 'window'.
```

Comment:
I believe it should also allow for Symbol params:

```ruby
driver.new_window(:tab)
driver.new_window(:window)
```

Reply:
@p0deje Why do you want it to automatically switch? That would provide less flexibility for the ruby libraries that use selenium (watir/capybara/etc)

Reply:
I'm not strongly supporting that, my arguments are mostly the same as Jim explained in [his comment](https://github.com/SeleniumHQ/selenium/issues/6873#issuecomment-459109624) and I'd prefer to keep consistency with other bindings (only C# supports it though).

What kind of flexibility wrappers need here? I can imagine a normal use case where a new window is created and context is switched to it immediately. What do I miss?

Reply:
👍 yes it does. I'll update documentation

Reply:
re:
>Why do you want it to automatically switch?

I agree consistency would be nice. Python also uses the switch_to format.

Would it be plausible to support both:
`driver.new_window(:tab)`
and
`driver.switch_to.new_window(:tab)`

Reply:
Another possible API might look like this:

```ruby
# you can do this
window_handle = driver.manage.new_window
driver.switch_to.window(window_handle)

# or you can do this
driver.switch_to.new_window
```

We already have window-related API in `driver.manage`, so providing 2 different ways to create new window might suit everyone.


@dylanlive Let's hear back from @twalpole before deciding which one to implement.

Reply:
The current window switching API is `driver.switch_to.window(handle)`, so if the new window API is only implemented via `driver.switch_to.new_window` then to create a new window and not switch to I have to do

    orig_handle = driver.window_handle
    driver.switch_to.new_window
    driver. switch_to.window(orig_handle)

which seems like extra overhead for something that should be relatively easy.  If the API in the current PR is used then to open a new window without switching it is 

    driver.new_window(:window)  # or driver.manage.new_window 

and to open and switch it's

    driver.switch_to.window(driver.new_window(:window)[:handle])

Adding the `driver.switch_to.new_window` as a shortcut for opening and switching would also be great, but implementing just that (and losing the ability to open without switching would mean Capybara resorting to calling another method directly on bridge (which it already does more times than I'd like)

Reply:
@twalpole Ok, but out of curiosity, what's the use case for opening a new window but not switching to it immediately?

Reply:
I don’t personally have one currently, but nothing else currently auto-switches context and I see no benefit to  preventing the creation without switching. A potential use case I could imagine is creating a pool of windows before a test and then using them in the tests, but that is purely hypothetical

Reply:
Alright, got it. It would be great to hear other maintainers opinions (cc @titusfortner @lmtierney). My call would be to prefer consistency across other language bindings over API flexibility, but maybe I'm just being too conservative here.

Reply:
@titusfortner @lmtierney Bump - any thoughts? I have a local version moving to the SwitchTo api that mimics other bindings.

I propose we mimic the other bindings, but I also think it'd be cool to have both options:
1. Automatically create a new window and switch using SwitchTo
2. Allow creating a new window, but not automatically switching to (providing flexibility for other frameworks)

Reply:
If we can do both I'd prefer that. Better to have flexibility, especially since the majority of users use a wrapping library. There are plenty of places ruby is different from other bindings so I wouldn't worry about that, this is at least similar

Reply:
Gah wrong button

Reply:
Gah wrong button

Reply:
Once we remove OSS backwards compatible hacks we won't have any code that makes multiple wire calls. I think watir / Capybara should decide if there's a more user friendly method required. I'd rather selenium not start supporting convenience methods.

Reply:
Could you not just add an extra default key to the input param

`driver.new_window(:window, switch: true)` And have it defaulted to whichever one is most desired / used in other bindings?

Reply:
Ok, so after talking to everyone, let's do what @twalpole and @titusfortner propose.

```ruby
driver.manage.new_window # defaults to tab
driver.manage.new_window(:tab)
driver.manage.new_window(:window)
```

it needs to return window handle that can be passed to `driver.window.switch_to`.

Reply:
@dylanlive Do you have time to make this change in the near future?

Reply:
Sounds good to me 👍 thanks all!

@twalpole I should have some time in the coming weeks